torch/_decomp/decompositions.py

```diff
@@ -3355,20 +3355,30 @@ def upsample_bilinear2d(
     v3 = aten._unsafe_index(input, [None, None, yp1, x])
     v4 = aten._unsafe_index(input, [None, None, yp1, xp1])
 
-    dtype = torch.float32 if not input.is_floating_point() else input.dtype
+    yscale = (y_f32 - y).clamp(0.0, 1.0).to(dtype)
+    xscale = (x_f32 - x).clamp(0.0, 1.0).to(dtype)
+
+    dtype = torch.int64 if not input.is_floating_point() else input.dtype
```

Comment:
`int64`? Given that the trick actually works with `int16`s (but we don't support `int16`s I don't think) it should be enough to use `int32`, which in turn we know how to vectorise on C++. We don't have vectorisation for double size types.